contrib/golang/filters/http/source/go/pkg/http/shim.go

```diff
@@ -44,35 +44,47 @@ import (
 
 var ErrDupRequestKey = errors.New("dup request key")
 
+var (
+	initOnce sync.Once
+	// each worker gets its own map which eliminates the need for a mutex
+	requests []map[*C.httpRequest]*httpRequest
```

Comment:
Adjusting `initOnce` and `requests` as member variables of `requestMap` would make it clearer.

Reply:
Done!